cmake/FindLIBHWLOC.cmake

```diff
@@ -14,8 +14,25 @@ find_file(LIBHWLOC_HEADER NAMES hwloc.h)
 get_filename_component(LIBHWLOC_INCLUDE_DIR ${LIBHWLOC_HEADER} DIRECTORY)
 set(LIBHWLOC_INCLUDE_DIRS ${LIBHWLOC_INCLUDE_DIR})
 
+if(LIBHWLOC_INCLUDE_DIR)
+    if(EXISTS "${LIBHWLOC_INCLUDE_DIR}/hwloc/autogen/config.h")
+        file(STRINGS "${LIBHWLOC_INCLUDE_DIR}/hwloc/autogen/config.h"
+             LIBHWLOC_VERSION
+             REGEX "#define[ \t]HWLOC_VERSION[ \t]\"[0-9]+.[0-9]+.[0-9]+\"")
+        string(
+            REGEX
+            REPLACE "#define[ \t]HWLOC_VERSION[ \t]\"([0-9]+.[0-9]+.[0-9]+)\""
+                    "\\1" LIBHWLOC_VERSION "${LIBHWLOC_VERSION}")
+    else()
+        message(
+            WARNING
+                "LIBHWLOC_INCLUDE_DIR found, but header with version info is missing"
+        )
+    endif()
+endif()
+
 if(WINDOWS)
-    find_file(LIBHWLOC_DLL NAMES "bin/hwloc-15.dll" "bin/libhwloc-15.dll")
+    find_file(LIBHWLOC_DLL NAMES "bin/hwloc-*.dll" "bin/libhwloc-*.dll")
```

Comment:
it looks like the dll file was not found...? perhaps the `*` is not working here as we expected it...?

Reply:
Yeah, it seems that `find_file` doesn't accept regex nor wildcards.